siriuspy/siriuspy/injctrl/main.py

```diff
@@ -1174,6 +1177,16 @@ def _thread_is_injecting(self):
         _time.sleep(self._isinj_duration/1000)
         self.run_callbacks('IsInjecting-Mon', _Const.IdleInjecting.Idle)
 
+    def _callback_havebeam(self, value, **kws):
+        if value:
+            return
+        if self._mode != _Const.InjMode.TopUp:
+            return
+        if self._topup_state_sts != _Const.TopUpSts.Off:
+            self._update_log('FATAL:We do not have stored beam!')
+            self._update_log('FATAL:Opening TopUp loop...')
+            self.set_topup_state(_Const.OffOn.Off)
```

Comment:
Wouldn't be better to run this on a separate thread?

Reply:
done!